src/actions/mover.ts

```diff
@@ -0,0 +1,339 @@
+      id: 'move',
+      weight: 8.5,
+    },
+  ];
+
+  /**
+   * Install the actions as both keyboard shortcuts and (where
+   * applicable) context menu items.
+   */
+  install() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.register(shortcut);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.register(menuItem);
+    }
+  }
+
+  /**
+   * Uninstall these actions.
+   */
+  uninstall() {
+    for (const shortcut of this.shortcuts) {
+      ShortcutRegistry.registry.unregister(shortcut.name);
+    }
+    for (const menuItem of this.menuItems) {
+      ContextMenuRegistry.registry.unregister(menuItem.id);
+    }
+  }
+
+  /**
+   * Returns true iff we are able to begin moving the given block (or,
+   * if no block supplied, the block wich currently has focus) on the
+   * given workspace.
+   *
+   * @param workspace The workspace to move on.
+   * @returns True iff we can beign a move.
+   */
+  canMove(workspace: WorkspaceSvg, block?: Block) {
+    if (!block) {
+      const cursor = workspace?.getCursor();
+      const curNode = cursor?.getCurNode();
+      block = curNode?.getSourceBlock() ?? undefined;
+    }
+
+    return !!(
+      this.navigation.getState(workspace) === Constants.STATE.WORKSPACE &&
+      this.canCurrentlyEdit(workspace) &&
+      !this.moves.has(workspace) && // No move in progress.
+      block?.isMovable()
+    );
+  }
+
+  /**
+   * Returns true iff we are currently moving a block on the given
+   * workspace.
+   *
+   * @param workspace The workspace we might be moving on.
+   * @returns True iff we are moving.
+   */
+  isMoving(workspace: WorkspaceSvg) {
+    return this.canCurrentlyEdit(workspace) && this.moves.has(workspace);
+  }
+
+  /**
+   * Start moving the currently-focused item on workspace, if
+   * possible.
+   *
+   * Should only be called if canMove has returned true.
+   *
+   * @param workspace The workspace we might be moving on.
+   * @returns True iff a move has successfully begun.
+   */
+  startMove(workspace: WorkspaceSvg, block?: Block) {
+    const cursor = workspace?.getCursor();
+    if (!block) {
+      const curNode = cursor?.getCurNode();
+      block = curNode?.getSourceBlock() ?? undefined;
+    }
+    if (!cursor || !block) throw new Error('precondition failure');
+
+    // Select and focus block.
+    common.setSelected(block as BlockSvg);
+    cursor.setCurNode(ASTNode.createBlockNode(block)!);
+
+    // Additional implementation goes here.
+    console.log('startMove');
+
+    this.moves.set(workspace, new MoveInfo(block));
+    return true;
+  }
+
+  /**
+   * Finish moving the currently-focused item on workspace.
+   *
+   * Should only be called if isMoving has returned true.
+   *
+   * @param workspace The workspace on which we are moving.
+   * @returns True iff move successfully finished.
+   */
+  finishMove(workspace: WorkspaceSvg) {
+    if (!workspace) return false;
```

Comment:
Make the `workspace` arg to this function nullable or remove this null check, same for `abortMove`. Or perhaps unify them since they're currently identical?